### DISCUSSION TITLE
Can I create multiple databases per project?

### DISCUSSION THREAD

**Discussion by bradCypress** [supabase/supabase#4471] [Questions] [answered]
I am unclear the relationship between projects and databases. I am looking to migrate from Azure Postgres to Supabase and my app is split among 5 different database. Do I need to create 5 projects each with their own database or is there a way to create multiple databases with a project? I use pgAdmin and it is not letting me add a new database - it shows up in the Database list but gives me a "unexpected response from login query" error trying to access it. And I see no where in the Supabase portal to view it.

**Comment by soedirgo** [comment, marked as answer]
At the moment you'd have to create 5 projects - creating multiple databases wouldn't work since all the other components (REST, Auth, Storage, etc.) are tied to a single database.

Edit: you can do [`CREATE DATABASE`](https://www.postgresql.org/docs/current/sql-createdatabase.html) to create multiple Postgres databases, but other components are tied to the `postgres` database.

**Reply by tananga** [reply]
Do you know if now in 2023 this is fixed?

**Reply by soedirgo** [reply]
The answer still applies, you should be able to run `CREATE DATABASE` to create additional databases. Does it not work for you?

**Reply by ghost** [reply]
Can confirm that I was able to use the `CREATE DATABASE` function on both a localhost supabase instance and a supabase-hosted project. Will any additional databases beyond the `postgres` database be backed up if we pay for the daily backups from supabase-hosted projects?

**Reply by soedirgo** [reply]
I believe it does, but you might want to verify this by [pausing & restoring](https://supabase.com/dashboard/project/_/settings/general) a test project.

**Reply by ghost** [reply]
@soedirgo good idea, I'll give that a shot. Thanks!

**Reply by bitcookr** [reply]
@soedirgo , I need to ensure that the shadow database has the PostGIS extension enabled.
I can do this by running ``CREATE EXTENSION postgis;`` query.
I created a new database named ``shadow_database`` using ``CREATE DATABASE shadow_database;``
But I'm not sure how can I switch into the database from primary one.

Does the current version support this?

**Reply by alexander-potemkin** [reply]
> @soedirgo good idea, I'll give that a shot. Thanks!

I wonder what was the outcome? Does the backup works as per your observations?

**Reply by dhanushreddy291** [reply]
> > @soedirgo good idea, I'll give that a shot. Thanks!
> 
> I wonder what was the outcome? Does the backup works as per your observations?

I have paused and resumed, it worked for me, there was still other db present

I guess only limitation would be supabase sdk related and they are tied to postgres but I think we can use other databases via any postgres compatible client
